src/realm/query_expression.hpp

```diff
@@ -2337,6 +2474,96 @@ class Columns<Link> : public Subexpr2<Link> {
     friend class Table;
 };
 
+template <>
+class Columns<SubTable> : public Subexpr2<SubTable> {
+public:
+    Columns(const Columns& other, QueryNodeHandoverPatches* patches)
+        : Subexpr2<SubTable>(other)
+        , m_link_map(other.m_link_map, patches)
+        , m_column_ndx(other.m_column_ndx)
+        , m_column(other.m_column)
+    {
+        if (m_column && patches)
+            m_column_ndx = m_column->get_column_index();
+    }
+
+    const Table* get_base_table() const override
+    {
+        return m_link_map.base_table();
+    }
+    void set_base_table(const Table* table) override
+    {
+        m_link_map.set_base_table(table);
+        m_column = &m_link_map.target_table()->get_column_table(m_column_ndx);
+    }
+
+    void verify_column() const override
+    {
+        m_link_map.verify_columns();
+        m_link_map.target_table()->verify_column(m_column_ndx, m_column);
+    }
+
+
+    std::unique_ptr<Subexpr> clone(QueryNodeHandoverPatches* patches) const override
+    {
+        return make_subexpr<Columns<SubTable>>(*this, patches);
+    }
+
+    void evaluate(size_t index, ValueBase& destination) override
+    {
+        Value<ConstTableRef>* d = dynamic_cast<Value<ConstTableRef>*>(&destination);
```

Comment:
evaluate() is the inner core of the query execution, so it's a bit unfortunate to have a slow dynamic_cast here